Mage/Logger/Writers/ConsoleWriter.cs

```diff
@@ -1,157 +1,201 @@
+			if (logEntry.Data != null)
+			{
+				UnityEngine.Debug.Log(logEntry.Data);
 			}
 		}
 
-		private void Info(object sender, LogEntry logEntry) {
-			UnityEngine.Debug.Log(makeLogString("info", logEntry.context, logEntry.message));
-			if (logEntry.data != null) {
-				UnityEngine.Debug.Log(logEntry.data);
+		private void Info(object sender, LogEntry logEntry)
+		{
+			UnityEngine.Debug.Log(makeLogString("info", logEntry.Context, logEntry.Message));
+			if (logEntry.Data != null)
+			{
+				UnityEngine.Debug.Log(logEntry.Data);
 			}
 		}
 
-		private void Notice(object sender, LogEntry logEntry) {
-			UnityEngine.Debug.Log(makeLogString("notice", logEntry.context, logEntry.message));
-			if (logEntry.data != null) {
-				UnityEngine.Debug.Log(logEntry.data);
+		private void Notice(object sender, LogEntry logEntry)
+		{
+			UnityEngine.Debug.Log(makeLogString("notice", logEntry.Context, logEntry.Message));
+			if (logEntry.Data != null)
+			{
+				UnityEngine.Debug.Log(logEntry.Data);
 			}
 		}
 
-		private void Warning(object sender, LogEntry logEntry) {
-			UnityEngine.Debug.LogWarning(makeLogString("warning", logEntry.context, logEntry.message));
-			if (logEntry.data != null) {
-				UnityEngine.Debug.LogWarning(logEntry.data);
+		private void Warning(object sender, LogEntry logEntry)
+		{
+			UnityEngine.Debug.LogWarning(makeLogString("warning", logEntry.Context, logEntry.Message));
+			if (logEntry.Data != null)
+			{
+				UnityEngine.Debug.LogWarning(logEntry.Data);
 			}
 		}
 
-		private void Error(object sender, LogEntry logEntry) {
-			UnityEngine.Debug.LogError(makeLogString("error", logEntry.context, logEntry.message));
-			if (logEntry.data != null) {
-				UnityEngine.Debug.LogError(logEntry.data);
+		private void Error(object sender, LogEntry logEntry)
+		{
+			UnityEngine.Debug.LogError(makeLogString("error", logEntry.Context, logEntry.Message));
+			if (logEntry.Data != null)
+			{
+				UnityEngine.Debug.LogError(logEntry.Data);
 			}
 		}
 
-		private void Critical(object sender, LogEntry logEntry) {
-			UnityEngine.Debug.LogError(makeLogString("critical", logEntry.context, logEntry.message));
-			if (logEntry.data != null) {
-				if (logEntry.data is Exception && (logEntry.data as Exception).StackTrace != null) {
-					Exception excpt = logEntry.data as Exception;
-					UnityEngine.Debug.LogError(excpt.ToString() + ":\n" + excpt.StackTrace.ToString());
-				} else {
-					UnityEngine.Debug.LogError(logEntry.data);
+		private void Critical(object sender, LogEntry logEntry)
+		{
+			UnityEngine.Debug.LogError(makeLogString("critical", logEntry.Context, logEntry.Message));
+			if (logEntry.Data != null)
+			{
+				if (logEntry.Data is Exception && (logEntry.Data as Exception).StackTrace != null)
+				{
+					var excpt = logEntry.Data as Exception;
+					UnityEngine.Debug.LogError(excpt + ":\n" + excpt.StackTrace);
+				}
+				else
+				{
+					UnityEngine.Debug.LogError(logEntry.Data);
 				}
 			}
 		}
```

Comment:
Though technically this is an improvement for later, the code annoyed me too.
So i cleaned it up a bit more. Its now basically what you have above, but centralised and now called from all the log functions to unify their treatment of data.